tools/@aws-cdk/cli-args-gen/lib/cli-type-gen.ts

```diff
@@ -129,9 +130,13 @@ function kebabToPascal(str: string): string {
     .join('');
 }
 
-function normalizeDefault(defaultValue: any): string {
-  if (typeof defaultValue === 'boolean' || typeof defaultValue === 'string') {
-    return String(defaultValue);
+function normalizeDefault(defaultValue: any, type: string): string {
+  const validDefaults = ['boolean', 'string', 'number', 'object'];
+  if (validDefaults.includes(typeof defaultValue)) {
+    return JSON.stringify(defaultValue);
   }
-  return 'undefined';
+
+  // We don't know what the default is, and only arrays get a special default
+  const generatedDefault = generateDefault(type);
+  return generatedDefault ? JSON.stringify(generatedDefault) : 'undefined';
```

Comment:
This feels like there must be a shorter way to write this, but I don't know what it is and this seems easy to understand so I'm fine with it

Reply:
I wrote it this way because of something funky: `JSON.stringify(undefined) = undefined` which is not the `'undefined'` string that i would want.